oauth/authcode.go

```diff
@@ -86,15 +87,20 @@ func (ac *AuthorizationCodeTokenSource) Token() (*oauth2.Token, error) {
 	}
 
 	verifier := base64.RawURLEncoding.EncodeToString(verifierBytes)
+	var url string
 
-	// Generate a code challenge. Only the challenge is sent when requesting a
-	// code which allows us to keep it secret for now.
-	shaBytes := sha256.Sum256([]byte(verifier))
-	challenge := base64.RawURLEncoding.EncodeToString(shaBytes[:])
+	if ac.ClientSecret != "" {
+		url = fmt.Sprintf("%s?response_type=code&client_id=%s&redirect_uri=http://localhost:8484/&scope=%s", ac.AuthorizeURL, ac.ClientID, strings.Join(ac.Scopes, `%20`))
```

Comment:
Perhaps we should make a utility to build these instead of repeating and slightly modifying the strings?